File: week2/time_file_write.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

int main(int argc, char *argv[]) {
    if (argc != 3) {
        printf("Usage: %s <number_of_lines> <output_file>\n", argv[0]);
        return 1;
    }

    int n = atoi(argv[1]);
    char *filename = argv[2];

    FILE *fp = fopen(filename, "w");
    if (fp == NULL) {
        perror("Error opening file");
        return 1;
    }

    clock_t start_time = clock();

    for (int i = 0; i < n; i++) {
        fprintf(fp, "This is line %d\n", i);
    }

    fclose(fp);

    clock_t end_time = clock();
    double run_time = (double)(end_time - start_time) / CLOCKS_PER_SEC;

    printf("Wrote %d lines to %s in %f seconds\n", n, filename, run_time);

    return 0;
}

File: week2/time_file_write.py
import sys
import time

if len(sys.argv) != 3:
    print(f"Usage: python3 {sys.argv[0]} <number_of_lines> <output_file>")
    sys.exit(1)

n = int(sys.argv[1])
filename = sys.argv[2]

start_time = time.time()

with open(filename, "w") as f:
    for i in range(n):
        f.write(f"This is line {i}\n")

end_time = time.time()
run_time = end_time - start_time

print(f"Wrote {n} lines to {filename} in {run_time:.6f} seconds")
